packages/reports/addon/components/navi-request-preview.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright 2019, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * Description: Navi Request Preview
+ *
+ * Usage:
+ *  <NaviRequestPreview
+ *    @request={{this.request}}
+ *    @visualization={{this.request}}
+ *    @onRemoveMetric={{action "oneRemoveMetric"}}
+ *    @onRemoveDimension={{action "onRemoveDimension"}}
+ *    @onRemoveTimeGrain={{action "onRemoveTimeGrain"}}
+ *    @onAddSort={{action "onAddSort"}}
+ *    @onRemoveSort={{action "onRemoveSort"}}
+ *  />
+ */
+import Component from '@ember/component';
+import layout from '../templates/components/navi-request-preview';
+import { layout as templateLayout, tagName } from '@ember-decorators/component';
+import { computed, get, set, action } from '@ember/object';
+import { capitalize } from '@ember/string';
+import { A as arr } from '@ember/array';
+import { inject as service } from '@ember/service';
+import { debounce } from '@ember/runloop';
+
+const ID_FIELD_MAP = {
+  metric: metric => metric.canonicalName,
+  dimension: dimension => dimension.dimension.name,
+  dateTime: () => 'dateTime'
+};
+
+const NEXT_SORT_DIRECTION = {
```

Comment:
maybe add a todo comment the next sort after `none` for dimensions should be `asc`

Reply:
Wait really? Why is that? As far as I understand, dimension sorting isn't currently supported in Navi at all.

Reply:
What I meant is - you have a couple of TODO comments for when dimension sorting is supported, I suggested adding one here as well under the assumption that sorting a dimension would be in an alphabetical order (asc).
Feel free to ignore this altogether.

Reply:
Ah I see. I was thinking we'd just follow the same order as metrics so that it's more consistent across each column.